File: src/main/java/io/zeitwert/ddd/validation/adapter/api/jsonapi/dto/AggregatePartValidationDto.java
package io.zeitwert.ddd.validation.adapter.api.jsonapi.dto;

import io.zeitwert.ddd.enums.adapter.api.jsonapi.dto.EnumeratedDto;
import io.zeitwert.ddd.validation.model.AggregatePartValidation;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AggregatePartValidationDto {

	private Integer seqNr;
	private EnumeratedDto validationLevel;
	private String validation;

	public static AggregatePartValidationDto fromValidation(AggregatePartValidation validation) {
		// @formatter:off
		return AggregatePartValidationDto.builder()
			.seqNr(validation.getSeqNr())
			.validationLevel(EnumeratedDto.fromEnum(validation.getValidationLevel()))
			.validation(validation.getValidation())
			.build();
		// @formatter:on
	}

}

File: fm-domain/src/main/java/io/zeitwert/dddrive/ddd/adapter/api/jsonapi/dto/AggregatePartValidationDto.kt
package io.zeitwert.dddrive.ddd.adapter.api.jsonapi.dto

import dddrive.app.validation.model.AggregatePartValidation

data class AggregatePartValidationDto(
	val seqNr: Int?,
	val validationLevel: EnumeratedDto?,
	val validation: String,
) {

	companion object {

		@JvmStatic
		fun fromValidation(validation: AggregatePartValidation): AggregatePartValidationDto =
			AggregatePartValidationDto(
				validation.seqNr,
				EnumeratedDto.of(validation.validationLevel),
				validation.message,
			)
	}

}
